src/plugins/data_view_editor/public/data_view_editor_service.ts

```diff
@@ -374,7 +378,7 @@ export class DataViewEditorService {
     );
 
     // necessary to get new observable value if the field hasn't changed
-    this.loadIndices();
+    await this.loadIndices();
```

Comment:
@mattkime there's a potential of a race condition if this loading takes longer then the following code, been adding an `await` here to catch this.